src/Interpreters/executeQuery.cpp

```diff
@@ -587,6 +604,117 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
 
             elem.current_database = context.getCurrentDatabase();
             elem.query = query_for_logging;
+            elem.normalized_query_hash = normalizedQueryHash(query_for_logging);
+
+            if (use_processors)
+            {
+                const auto & info = context.getQueryAccessInfo();
+                elem.query_databases = info.databases;
+                elem.query_tables = info.tables;
+                elem.query_columns = info.columns;
+            }
+
+            if (!query_database.empty() && query_table.empty())
+            {
+                elem.query_databases.insert(query_database);
+            }
+            else if (!query_table.empty())
+            {
+                if (query_database.empty())
+                    query_database = context.getCurrentDatabase();
+                elem.query_databases.insert(query_database);
+                elem.query_tables.insert(query_database + "." + query_table);
+            }
+
+            {
```

Comment:
The code is too complicated, maybe add a virtual method to IInterpreter instead?

Reply:
Done.